### ISSUE TITLE
warning C4706 - Visual Studio 2017 (/W4)

### ISSUE THREAD

**Issue by jseward** [nlohmann/json#784]
```
json.hpp(3037): warning C4706: assignment within conditional expression
json.hpp(3133): warning C4706: assignment within conditional expression
```

It looks like this was discussed and solved already for VS 2015 here : https://github.com/nlohmann/json/issues/295

But the issue remains on VS 2017

**Comment by nlohmann** [comment]
The assignment is OK here, and it is surrounded by parentheses which silences GCC and Clang. Maybe, adding pragmas could help here. I have to check #295 again.

**Comment by nlohmann** [comment]
Fixed with #785
